File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        hm = Counter(nums)

        heap = []
        for key, freq in hm.items():
            heapq.heappush(heap, (-freq, key))
        
        res = []
        for _ in range(k):
            freq, num = heapq.heappop(heap)
            res.append(num)

        return res

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        //[3, 2, 4, 6]
        //[48, 24, 6, 1]
        //[48, 72, 36, 24]

        int[] right = new int[nums.length];

        right[nums.length - 1] = 1;
        for (int i = nums.length - 2; i >= 0; i--) {
            right[i] = right[i + 1] * nums[i + 1];
        }

        int[] res = new int[nums.length];
        int i = 0;
        int product = 1;
        while (i < nums.length) {
            res[i] = product * right[i];
            product *= nums[i];
            i++;
        }

        return res;
    }
}  
